src/components/Table/TableFormatters/DiscoverConfirmFormatter.vue

```diff
@@ -123,19 +123,19 @@ export default {
         {
           name: 'delete_remote',
           label: this.$t('DeleteRemoteAccount'),
-          has: this.row.remote_present,
+          has: this.row.remote_present && this.$hasPerm('accounts.remove_account'),
           disabled: this.$store.getters.currentOrgIsRoot
         },
         {
           name: 'add_account',
           label: this.$t('AddAccount'),
-          has: !this.row.present,
+          has: !this.row.present && this.$hasPerm('accounts.add_account'),
           disabled: this.$store.getters.currentOrgIsRoot
         },
         {
           name: 'change_password_add',
           label: this.$t('AddAccountAfterChangingPassword'),
-          has: !this.row.present,
+          has: !this.row.present && this.$hasPerm('accounts.add_pushaccountexecution'),
           disabled: this.$store.getters.currentOrgIsRoot
         }
       ]
```

Comment:
The only issue I can see is the use of `!` before some boolean expressions. This could cause errors if values other than false are used.

In terms of optimization, you might want to refactor the array structure so it's easier to maintain and edit later on:

```
export default [
   {
       // ...  
   },
{
   "name": "delete_remote",
    ...
},
...
]
```